econml/dml/causal_forest.py

```diff
@@ -46,11 +65,93 @@ def fit(self, X, T_res, Y_res, sample_weight=None, sample_var=None):
         if Y_res.ndim == 1:
             Y_res = Y_res.reshape((-1, 1))
         self._model.fit(fts, T_res, Y_res, sample_weight=sample_weight)
+        # Fit a doubly robust average effect
+        if self._discrete_treatment and self._drate:
+            oob_preds = self._model.oob_predict(fts)
+            self._oob_preds = oob_preds
+            if np.any(np.isnan(oob_preds)):
+                warn("Could not generate out-of-bag predictions on some training data. "
+                     "Consider increasing the number of trees. `ate_` results will take the "
+                     "average of the subset of training data for which out-of-bag predictions "
+                     "where available.")
+            residuals = Y_res - np.einsum('ijk,ik->ij', oob_preds, T_res)
+            propensities = T - T_res
+            VarT = propensities * (1 - propensities)
+            drpreds = oob_preds
+            drpreds += cross_product(residuals, T_res / VarT).reshape((-1, Y_res.shape[1], T_res.shape[1]))
+            drpreds[np.isnan(oob_preds)] = np.nan
+            self.ate_, self.ate_stderr_ = self._ate_and_stderr(drpreds)
+            self.att_ = []
+            self.att_stderr_ = []
+            att, stderr = self._ate_and_stderr(drpreds, np.all(T == 0, axis=1))
+            self.att_.append(att)
+            self.att_stderr_.append(stderr)
+            for t in range(self._d_t[0]):
+                att, stderr = self._ate_and_stderr(drpreds, (T[:, t] == 1))
+                self.att_.append(att)
+                self.att_stderr_.append(stderr)
+
         return self
 
     def predict(self, X):
         return self._model.predict(self._combine(X, fitting=False)).reshape((-1,) + self._d_y + self._d_t)
 
+    @property
+    def ate_(self):
+        if not self._discrete_treatment:
+            raise AttributeError("Doubly Robust ATE calculation on training data "
+                                 "is available only on discrete treatments!")
+        if not self._drate:
+            raise AttributeError("Doubly Robust ATE calculation on training data "
+                                 "is available only when `drate=True`!")
+        return self._ate

```

Comment:
For some reason the coverage report shows this line (and the corresponding `ate_stderr_` one as uncovered...  Does that make sense?